lib/server.go

```diff
@@ -2000,9 +2118,30 @@ func (srv *Server) _handleBlock(pp *Peer, blk *MsgDeSoBlock) {
 		return
 	}
 
-	// If we get here, it means we're in SyncStateFullySynced, which is great.
+	// If we get here, it means we're in SyncStateFullyCurrent, which is great.
 	// In this case we shoot a MEMPOOL message over to the peer to bootstrap the mempool.
 	srv._maybeRequestSync(pp)
+
+	///////////////////// PoS Validator Consensus Initialization /////////////////////
+
+	// Exit early if the chain isn't SyncStateFullyCurrent.
+	if srv.blockchain.chainState() != SyncStateFullyCurrent {
+		return
+	}
+
+	// Exit early if the current tip height is below the PoS cutover height. We are ready to enable
+	// the PoS validator FastHotStuffConsensus once we reach the final block of the PoW protocol.
+	// This requires the block height check to use ProofOfStake2ConsensusCutoverBlockHeight-1
+	tipHeight := srv.blockchain.blockTip().Height
+	if tipHeight < srv.blockchain.params.ForkHeights.ProofOfStake2ConsensusCutoverBlockHeight-1 {
+		return
+	}
+
+	// If the PoS validator FastHotStuffConsensus is initialized but not yet running, then
+	// we can start the validator consensus, and transition to it in the steady-state.
+	if srv.fastHotStuffConsensus != nil && !srv.fastHotStuffConsensus.IsRunning() {
+		srv.fastHotStuffConsensus.Start()
```

Comment:
Just a heads-up. By the time we get here, we're guaranteed that our tip will be within MaxTipAge of the "true" tip on the network, where MaxTipAge is 24h. Specifically, the order is: 1) sync all the headers, 2) download all the blocks for those headers, 3) the next thing that would trigger with your code is _handleBlock().

If we assume the above, then we could potentially be an hour or more behind at step (2) causing the first tip block we get to do a lot of catch-up. I think this is probably fine, since an hour of blocks is only 3600 little blocks. But it's something to keep in mind as your testing. And if it's slow or suboptimal, you can do the update to add your committed tip to GetBlocks() that I mentioned previously.

Reply:
The only caveat is that depending on how far back the timed out node is it, the syncing may be inefficient. It's the same inefficiency you pointed out in https://github.com/deso-protocol/core/pull/875#discussion_r1460183510

I think that's an optimization we can defer for the future depending on testing looks on devnet